arelle/plugin/validate/NL/rules/fr_kvk.py

```diff
@@ -82,6 +86,56 @@ def rule_fr_kvk_2_01(
                 )
 
 
+def etree_iter(node: ModelObject | _Element) -> Any:
+    yield node
+    for child in node.iterchildren():
+        for n_d in etree_iter(child):
+            yield n_d
+
+
+@validation(
+    hook=ValidationHook.XBRL_FINALLY,
+    disclosureSystems=[
+        DISCLOSURE_SYSTEM_NT16,
+        DISCLOSURE_SYSTEM_NT17,
+    ],
+)
+def rule_fr_kvk_2_02(
+        pluginData: PluginValidationDataExtension,
+        val: ValidateXbrl,
+        *args: Any,
+        **kwargs: Any,
+) -> Iterable[Validation] | None:
+    """
+    FR-KVK-2.02: The attribute 'xml:lang' MUST contain the same value within an XBRL instance document
+    """
+    modelXbrl = val.modelXbrl
+    docXmlLang = []
+    for fact in modelXbrl.facts:
+        if fact.xmlLang and fact.xmlLang not in docXmlLang:
+            docXmlLang.append(fact.xmlLang)
+    assert modelXbrl.modelDocument is not None
+    if modelXbrl.modelDocument.type in (ModelDocument.Type.INLINEXBRL, ModelDocument.Type.INLINEXBRLDOCUMENTSET, ModelDocument.Type.UnknownXML):
+        for ixdsHtmlRootElt in modelXbrl.ixdsHtmlElements:
+            for elt in etree_iter(cast(_Element, ixdsHtmlRootElt)):
+                if isinstance(elt, ModelInlineFootnote) or (isinstance(elt, ModelResource) and elt.qname == XbrlConst.qnLinkFootnote):
+                    if elt.xmlLang and elt.xmlLang not in docXmlLang:
+                        docXmlLang.append(elt.xmlLang)
+    elif modelXbrl.modelDocument.type == ModelDocument.Type.INSTANCE:
+        for elt in modelXbrl.modelDocument.xmlRootElement.iter():
```

Comment:
This is skipping over the root element itself, which is supposed to define what the language is for the entire document. See FR-KVK-2.01. I think a preferred approach would be to retrieve the root language first, and `yield` a validation error for every instance of a mismatching language within the doc, that way the error can include document and line number for each occurrence.